providers/provider.go

```diff
@@ -0,0 +1,33 @@
+package providers
+
+import (
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/argoproj/argo-rollouts/pkg/apis/rollouts/v1alpha1"
+	"github.com/argoproj/argo-rollouts/providers/prometheus"
+)
+
+// Provider methods to query a external systems and generate a measurement
+type Provider interface {
+	// Run start a new external system call for a measurement
+	// should be idoponent and do nothing is a call has been started
```

Comment:
idempotent and do nothing if a call has been started

Reply:
done